### ISSUE TITLE
SA1108: should allow comments on the line that precedes the block

### ISSUE THREAD

**Issue by AArnott** [DotNetAnalyzers/StyleCopAnalyzers#992]
I propose that the following be allowed by SA1108:

```csharp
    if (condition) // comment
    {
    }
```

[The docs for SA1108](http://www.stylecop.com/docs/SA1108.html) show examples of violation which I agree with. I don't know if the original stylecop permitted the above syntax or not -- I suspect it did, simply because I don't remember this being disallowed before and it's a syntax I like.

**Comment by sharwell** [comment]
I'll put in a vote for not allowing a comment to ever share a line with non-trivia tokens.

**Comment by AArnott** [comment]
AFAIK there is no stylecop rule against appending single line comments to the end of a line of code. So if that's your goal, that should be a new rule and could be enforced in a case like this too, independently of how we decide to handle SA1108.

StyleCop's rule that insists that single-line comments (that stand on a line by themselves) are preceded by a blank line make it particularly burdensome to move this comment to its own line because then I have to separate this code from the code above it, yet I often don't want to do that because it's logically one operation. I reserve comments that share a line with code to those that are very short and explanatory of something on that line in particular. I use the blank line plus dedicated comment line approach for explaining the logical unit of code I'm about to execute.

**Comment by sharwell** [comment]
If a new separate rule was available to enforce no comment sharing, then I would probably be :ok: with rewording SA1108 to specifically disallow a comment on the line between `)` and `{` of a compound statement.

**Comment by sharwell** [comment]
:memo: Now that we have a configuration file, a new option would suffice without requiring a new rule.
